src/header/internals/cell.rs

```diff
@@ -87,6 +87,20 @@ impl<V: ?Sized + Any + 'static> PtrMapCell<V> {
     }
 
     #[inline]
+    pub fn into_value(self, key: TypeId) -> Option<Box<V>> {
+        let mut map = unsafe { self.0.into_inner() };
```

Comment:
Also `UnsafeCell::get` is not marked as unsafe, I don't understand why it is wrapped in an `unsafe` block in `get` and `get_mut`.